src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -79,19 +79,20 @@ public async Task<EmissionsForecast> GetCarbonIntensityForecastAsync(Location lo
         using (var activity = Activity.StartActivity())
         {
             var balancingAuthority = await this.GetBalancingAuthority(location, activity);
-            var forecast = await this.WattTimeClient.GetForecastOnDateAsync(balancingAuthority, TimeToLowestIncrement(requestedAt));
+            var roundedRequestedAt = TimeToLowestIncrement(requestedAt);
+            var forecast = await this.WattTimeClient.GetForecastOnDateAsync(balancingAuthority, roundedRequestedAt);
             if (forecast == null)
             {
-                Exception ex = new ArgumentException($"No forecast was generated at the requested time {requestedAt}");
+                var ex = new ArgumentException($"No forecast was generated at the requested time {roundedRequestedAt}");
                 activity?.SetStatus(ActivityStatusCode.Error, ex.Message);
                 Logger.LogError(ex, ex.Message);
                 throw ex;
             }
-            return ForecastToEmissionsForecast(forecast, location); 
+            return ForecastToEmissionsForecast(forecast, location, roundedRequestedAt); 
```

Comment:
I think this should be the original `requestedAt` that the user passes.  The rounding is an implementation detail for how we retrieve the correct forecast, but not relevant to the user.  Especially since it is rounded specifically to match the `generatedAt` field that they will also be getting in the payload.

Reply:
oh ok, if that is the intent, cool, I'm good too.